### ISSUE TITLE
Flow polymorphic types cannot be parsed

### ISSUE THREAD

**Issue by aaronjensen** [reactjs/react-docgen#125]
I had this:

```js
export type Props = {
  handleSubmit: () => void,
  onBack?: () => void,
  children?: React.Element<*>,
  backText?: string,
  nextText?: string,
}
```

And it failed to parse so I had to change my children type to `mixed`. I'm guessing this has to do w/ the fact that docgen is using an old version of babylon.

**Comment by fkling** [comment]
Yeah, I think we should simply use the flow parser since it's more ESTree-compatible than babylon 6.

**Comment by aaronjensen** [comment]
babylon 6 can handle all of these though, right? Is it just a version issue?

The flow parser is ocaml, so I'm not sure it could be used here.

**Comment by fkling** [comment]
There is JS version of the flow parser available: https://www.npmjs.com/package/flow-parser . I'd rather use flow since I know they are trying to stay close to the ESTree spec.
